src/material.js

```
@@ -172,23 +172,59 @@ export class Material {
+     * @returns {Object}
+     */
+    getDerivedPropertyByName(name) {
+        return this.getDerivedProperties().find(x => x.name === name);
+    }
+
+    /**
+     * Calculates the hash for a material.
+     * If `isNonPeriodic` = false
+     * The hash is calculated from the basis and lattice. The algorithm expects the following:
      * - asserts lattice units to be angstrom
      * - asserts basis units to be crystal
      * - asserts basis coordinates and lattice measurements are rounded to hash precision
      * - forms strings for lattice and basis
      * - creates MD5 hash from basisStr + latticeStr + salt
+     *
+     * If `isNonPeriodic` = true
+     * The hash based on the `inchi` string for the material that is returned from getNonPeriodicHashMessage.
+     *
      * @param salt {String} Salt for hashing, empty string by default.
      * @param isScaled {Boolean} Whether to scale the lattice parameter 'a' to 1.
      */
     calculateHash(salt = '', isScaled = false) {
-        const message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        let message;
+        if (!this.isNonPeriodic) {
+            message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        } else {
+            message = this.getNonPeriodicHashMessage();
+        }
         return CryptoJS.MD5(message).toString();
     }
 
-    get hash() {
-        return this.calculateHash();
+    /**
+     * Gets the `inchi` string from the derived properties and returns it as the message to use for
+     * generating the hash when `isNonPeriodic` = true.
+     * @returns String
+     */
+    getNonPeriodicHashMessage() {
+        const inchiString = this.getDerivedProperties() ? this.getDerivedPropertyByName('inchi') : null;
+        if (inchiString) {
+            return inchiString.value;
+        } else {
+            throw new Error("Non-Periodic hash could not be created. Missing InChI.");
```

updated to throw error instead of logging it